### PR TITLE
Do not use more than one file descriptor at a time

### LINKED ISSUE #994: Too many files open error with prettier cli
I accidentally passed 10000 files to prettier which makes it crash on my Mac because it opens too many files in parallel.

See https://github.com/prettier/prettier/blob/master/bin/prettier.js#L214

Here are two solutions:

* Turn this into `fs.readFileSync`. All of prettiers code is CPU bound. The overhead of reading all files asynchronously actually makes prettier slower here.
* Throttle the FS operations or use graceful-fs.

Alternatively: add a way to actually exclude node_modules from the glob pattern. I couldn't make it work so I had to do my own filtering before calling prettier.

### PR DESCRIPTION
**Pull request by vjeux**
@cpojer made a great suggestion to use `fs.readFileSync` instead of the async version. prettier is CPU-bound and if you want parallelism you can spin up many instances of prettier.

Fixes #994

### CODE REVIEW

File: bin/prettier.js [hunk 1]
```diff
@@ -210,71 +210,72 @@ if (stdin) {
...
   eachFilename(filepatterns, filename => {
-    fs.readFile(filename, "utf8", (err, input) => {
-      if (write || argv["debug-check"]) {
-        // Don't use `console.log` here since we need to replace this line.
-        process.stdout.write(filename);
-      }
+    if (write || argv["debug-check"]) {
+      // Don't use `console.log` here since we need to replace this line.
+      process.stdout.write(filename);
+    }
```
Comment (vjeux) [inline bin/prettier.js]: I just moved this outside of the callback

File: bin/prettier.js [hunk 2]
```diff
@@ -210,71 +210,72 @@ if (stdin) {
...
+      input = fs.readFileSync(filename, "utf8");
+    } catch(e) {
+      // Add newline to split errors from filename line.
+      process.stdout.write("\n");
+
+      console.error("Unable to read file: " + filename + "\n" + err);
+      // Don't exit the process if one file failed
+      process.exitCode = 2;
+      return;
+    }
```
Comment (vjeux) [inline bin/prettier.js]: and added a try catch here. All the rest is unindented once.

**Comment by vjeux** [comment]
This may (or may not) help with https://github.com/prettier/prettier/issues/980

**Comment by jlongster** [comment]
Sounds good to me. Should we also be careful about traversing into `node_modules`?

**Comment by vjeux** [comment]
We may want to disable it by default with a warning and add a --go-through-node-modules flag to bypass it if we really care about preventing people from shooting themselves in the foot.

**Pull request merged**